manager/dispatcher/assignments.go

```diff
@@ -46,9 +35,24 @@ func (a *assignmentSet) addTaskDependencies(readTx store.ReadTx, t *api.Task) {
 	if container != nil {
 		secrets = container.Secrets
 	}
+
+	for _, resourceRef := range t.Spec.ResourceReferences {
+		a.log.WithField("resource.id", resourceRef.ResourceID)
+		mapKey := typeAndID{objType: resourceRef.ResourceType, id: resourceRef.ResourceID}
+		if len(a.tasksUsingDependency[mapKey]) == 0 {
+			a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+			resource := store.GetResource(readTx, resourceRef.ResourceID)
+			if resource == nil {
+				a.log.Debug("resource not found")
+				continue
+			}
+		}
+		a.tasksUsingDependency[mapKey][t.ID] = struct{}{}
```

Comment:
ah this explains a lot about why I couldn't understand the mismatched values that test for changes.  Thanks!